core/src/main/java/org/apache/metamodel/convert/Converters.java

```diff
@@ -188,7 +191,7 @@ private static void autoDetectConvertersInternally(DataContext dataContext, Tabl
             detectors.put(column, new ColumnTypeDetector());
         }
 
-        Query query = dataContext.query().from(table).select(columns).toQuery();
+        Query query = dataContext.query().from(table).select(columns.toArray(new Column[columns.size()])).toQuery();
```

Comment:
Same extra toArray here (just marking them in case I'm right that we can avoid them).